src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -509,6 +513,148 @@ private OidcConfiguration buildOidcConfiguration() {
         return conf;
     }
 
+    private static void filterJwsAlgorithms(OIDCProviderMetadata oidcProviderMetadata) {
```

Comment:
NIT: this class is already getting quire big and as this is static these methods could be moved to a utility class.